### PR TITLE
Feature/all items

### PR DESCRIPTION
**Pull request by serpient**
# Creates All Items Page
All Items Page uses a new fetch utility to make an API call for items data. The data is then set using ReactHooks, rendered as ItemCard components, then placed as child components in the ItemsPagePresenter component.

- Create fetch utility that will make a get request given a endpoint. Can be used standalone, or in a useEffect() call in the following below example format. As a hint, when using useEffect(), be sure to set the second argument to determine when it should execute. Not setting it correctly will cause the function trigger numerous times. Read more here -> https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
```
const [items, setItems] = useState([]);

    useEffect(() => {
        const fetchItems = async () => {
            let result = await Fetch("/api/items");
           setItems(result)
        }
        fetchItems();
    }, []);
```
- Add tests for ItemsPageContainer using axiosMock and testing-library/react. Currently only asserting that the correct number of data objects will be rendered.
- Create starter shared scss variables file
- Create UI Page modular component
- Each component is split into Container (logic) and Presenter (presentational).

# Things To Do
- Add edge case tests for Error / Loading states in All Items Page & ItemCard

# Live Shot
Screen Shot 2019-07-17 at 11 02 56 PM

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,7 +12,6 @@ function App() {
       <Router>
         <Route path="/" exact component={ItemsPage} />
         <Route path="/items" exact component={ItemsPage} />
-        <Route path="/item/:id" exact component={ItemPage} />
       </Router>
```
Comment (tpetersen0308) [inline src/App.js]: This route should be `/items/:id`

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (zacholauson) [inline .vscode/launch.json]: I don't know how vscode functions, but do we need both `launch.json` files? Also, JSON doesn't have a valid comments format so they should probably be removed from there.

File: src/__mocks__/axios.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const axios = jest.genMockFromModule('axios');
+
+const mockAxios = {
```
Comment (zacholauson) [inline src/__mocks__/axios.js]: https://8thlight.com/blog/eric-smith/2011/10/27/thats-not-yours.html
https://github.com/testdouble/contributing-tests/wiki/Don't-mock-what-you-don't-own

File: src/components/items/Presenter.test.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ItemsPagePresenter from './Presenter';
+
+it('renders without crashing', () => {
```
Comment (zacholauson) [inline src/components/items/Presenter.test.js]: Should we be asserting something about how `Presenter` renders its children?

File: .vscode/launch.json (hunk 2 above)
Comment (jayden) [reply .vscode/launch.json]: As an aside, should `.vscode` be ignored completely?

File: src/components/items/Container.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        const fetchItems = async () => {
+            let result = await Fetch("/api/items");
+            let { data, error } = result;
+            data ? setItems(data) : setError(error);
+        }
+        fetchItems();
+    }, []);
+
+    const createItemComponents = (data) => {
+        return data.map((object, index) => {
```
Comment (jayden) [inline src/components/items/Container.js]: It's an [anti-pattern](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318) to use the index when creating component lists. I'd suggest using some other unique identifier like the item's id.

File: src/utilities/fetch.test.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    };
+
+    axiosMock.get.mockResolvedValueOnce(itemData)
+    const url = '/api/items';
+    const fetchResult = await Fetch(url);
+
+    expect.assertions(3);
+    expect(axiosMock.get).toHaveBeenCalledTimes(1)
+    expect(axiosMock.get).toHaveBeenCalledWith(url)
+    expect(fetchResult).toEqual(itemData.data)
```
Comment (zacholauson) [inline src/utilities/fetch.test.js]: This doesn't seem like an overly useful test since it's really just testing that the mock returns what you told it to return.

File: src/components/item_card/Presenter.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import './Presenter.scss';
+
+const ItemCard = ({ item }) => {
+    let { title, price, image, end_date } = item;
+    const dateFormatter = (endDate) => {
```
Comment (zacholauson) [inline src/components/item_card/Presenter.js]: Are there any tests for this?

Comment (serpient) [reply src/components/item_card/Presenter.js]: We extracted this out and added tests in the `show_item` PR

File: src/__mocks__/axios.js (hunk 3 above)
Comment (serpient) [reply src/__mocks__/axios.js]: I refactored to create a APIHandler that can take in a Fetcher and MockFetcher class

**Review by zacholauson — APPROVED** [review]
